src/units/abilities.cpp

```diff
@@ -854,6 +854,31 @@ std::string attack_type::weapon_specials(bool only_active, bool is_backstab) con
 		}
 	}
 
+	assert(display::get_singleton());
+	const unit_map& units = display::get_singleton()->get_units();
+	if(self_){
+		std::set<std::string> checking_name;
+		const auto adjacent = get_adjacent_tiles(self_loc_);
+		for(unsigned i = 0; i < adjacent.size(); ++i) {
+			const unit_map::const_iterator it = units.find(adjacent[i]);
+			if (it == units.end() || it->incapacitated())
+				continue;
+			for (const config::any_child &sp : (*it).abilities().all_children_range()){
+				const bool active =
+					(&*it == self_.get())
+					? check_self_abilities_impl(shared_from_this(), other_attack_, sp.cfg, self_, self_loc_, AFFECT_EITHER, sp.key)
+					: check_adj_abilities_impl(shared_from_this(), other_attack_, sp.cfg, self_, *it, i, self_loc_, AFFECT_EITHER, sp.key);
+
+				const std::string& name = active ? sp.cfg["name"].str() : "";
+
+				if (!name.empty() && checking_name.count(name) == 0) {
+					checking_name.insert(std::move(name));
+					if (!res.empty()) res += ", ";
+					res += name;
```

Comment:
Does appending something after it's been `move`d work as it's supposed to here?  I thought the result of a `move` was that the object would be left in an unknown but valid state.

Reply:
move removed